vision/use-cases/video-thumbnail-generation/video_thumbnail_generation.ipynb

```diff
@@ -0,0 +1,804 @@
+        {
+          "name": "stdout",
+          "output_type": "stream",
+          "text": [
+            "```json\n",
+            "{\n",
+            "    \"timestamp\": [\n",
+            "        \"00:00\",\n",
+            "        \"00:13\",\n",
+            "        \"00:37\",\n",
+            "        \"01:00\"\n",
+            "    ],\n",
+            "    \"caption\": [\n",
+            "        \"A woman is packing for a trip.\",\n",
+            "        \"The woman is looking for her hotel.\",\n",
+            "        \"The woman is at the hotel reception.\",\n",
+            "        \"The woman is visiting a park.\"\n",
+            "    ]\n",
+            "}\n",
+            "```\n"
+          ]
+        }
+      ],
+      "source": [
+        "print(response_text)"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "ec1c760d511b"
+      },
+      "source": [
+        "### displaying thumbnail results with captions"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "54fa92eafb2a"
+      },
+      "outputs": [],
+      "source": [
+        "display_results(response_text, clip)"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "c8d722df9b21"
+      },
+      "source": [
+        "# Case 2: Using an Advanced Prompt"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "dddce2bb59b7"
+      },
+      "source": [
+        "### Writing an advanced prompt for better thumbnail generation"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": 53,
+      "metadata": {
+        "id": "6aa02b9c3527"
+      },
+      "outputs": [],
+      "source": [
+        "advanced_prompt = \"\"\"You are an expert in video content creation and content marketing.\n",
+        "You have the ability to find best thumbnails from a video and provide meaningful and short and catchy captions for them.\n",
+        "Your task is to find the best 4 thumbnails from a given video along with short, and meaningful captions that is good for marketing.\n",
+        "Consider the following rules while generating thubmnails:\n",
+        "\n",
+        "- Thumbnail should have clear focus on the key objects and people, less focus on background\n",
+        "- Thumbnail image should be high quality and bright, avoid blurry images\n",
+        "- Thumbnail image and caption together tell a story\n",
+        "- Thumbnail caption is good for marketing\n",
+        "\n",
+        "Provide response in a proper JSON format similar to following:\n",
+        "{\n",
+        "    \\\"timestamp\\\": [],\n",
+        "    \\\"caption\\\": []\n",
+        "}\n",
+        "\"\"\""
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "1e718c0ce536"
+      },
+      "source": [
+        "### calling Gemini API"
+      ]
```

Comment:
Not sure if these extra headers are needed

Reply:
updated with more meaningful header